### PR TITLE
[Java/helidon]: first attempt at introducing Helidon

### PR DESCRIPTION
**Pull request by whiplash**

### CODE REVIEW

File: java/helidon/config.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+build:
+  - mvn package
+
+files:
+  - target/helidon.webserver.jar
+
+binaries:
+  - target/helidon.webserver.jar
+
+command: /usr/bin/java -XX:-RelaxAccessControlCheck -XX:+UseNUMA -XX:+UseParallelGC -XX:+AggressiveOpts -jar /opt/web/target/helidon.webserver.jar
```
Comment (waghanza) [inline java/helidon/config.yaml:14]: `AggressiveOpts` does not exists anymore

**Review by waghanza — CHANGES_REQUESTED** [review]
when running app on container

```
Error: Unable to initialize main class web.helidon.Main
Caused by: java.lang.NoClassDefFoundError: io/helidon/media/common/MediaSupport
```

File: java/helidon/config.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+framework:
+  website: helidon.io
+  version: 3.0.1
```
Comment (waghanza) [inline java/helidon/config.yaml:3]:
```suggestion
  version: 3.0
```

File: java/helidon/pom.xml [hunk 3]
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>io.helidon.applications</groupId>
+        <artifactId>helidon-se</artifactId>
+        <version>3.0.1</version>
```
Comment (waghanza) [inline java/helidon/pom.xml:10]:
```suggestion
        <version>[3.0,3.1)</version>
```

**Comment by whiplash** [comment]
not sure what's going on here

**Comment by waghanza** [comment]
it run inside the first layer, seems to be something not copied between containes

**Comment by whiplash** [comment]
anything I can try to fix this?

**Comment by waghanza** [comment]
maybe creating a all-in-one jar, honestly not sure to remember the name, fatjar, uberjar ... the idea is to have one jar file without extra dependencies.

the jar is not working since some required library are not copied from build container

**Comment by whiplash** [comment]
> the idea is to have one jar file 

this is what `mvn package` does

**Comment by waghanza** [comment]
then this issue comes from `pom.xml`, I mean a class could not be found
`Caused by: java.lang.NoClassDefFoundError: io/helidon/media/common/MediaSupport`

try :
- bundle exec rake config
- cd java/helidon
- docker build . -f .Dockerfile -t test
- docker run -ti test ash
- /usr/bin/java -XX:-RelaxAccessControlCheck -XX:+UseNUMA -XX:+UseParallelGC -jar /opt/web/target/helidon.webserver.jar

**Comment by waghanza** [comment]
still an error
```
Error: Unable to initialize main class web.helidon.Main
Caused by: java.lang.NoClassDefFoundError: io/helidon/media/common/MediaSupport
```

**Comment by whiplash** [comment]
let's see if @tomas-langer or @barchetta can help

**Comment by barchetta** [comment]
I pulled this PR and I can build and run the helidon application in a shell.  My hunch is you are not copying `target/libs` to your container image. 

Our default packaging consists of:

1. `target/application.jar`: this contains the application code and a `Class-Path` entry for runtime dependencies
2. `target/libs`: contains all runtime dependencies for the application as referenced by the jar's `Class-Path`

**Comment by barchetta** [comment]
@whiplash we consider the uber-jar an anti-pattern as they are not Docker image friendly and they introduce complexities concerning merging of resources (like `META-INF/services/*`). 

Another thing to note is that we support packaging the application as a jlink runtime image.  I have verified that this works with this PR:

```
mvn clean install -Pjlink-image
target/helidon.webserver-jri/bin/start
```

In this case `target/helidon.webserver-jri` contains a custom Java Runtime Image, plus the application code, plus the runtime dependencies.

**Comment by whiplash** [comment]
Thanks so much @barchetta. 

@waghanza we can `try` with this jlink runtime and see how it goes

**Comment by whiplash** [comment]
@waghanza maybe we made it :)

**Review by waghanza — APPROVED** [review]

**Pull request merged**